cpp/src/parquet/metadata.cc

```diff
@@ -894,6 +894,13 @@ class FileMetaData::FileMetaDataImpl {
     out->impl_->key_value_metadata_ = key_value_metadata_;
     out->impl_->file_decryptor_ = file_decryptor_;
 
+    if (key_value_metadata_ && key_value_metadata_->Contains("row_group_aad_0")) {
+      PARQUET_ASSIGN_OR_THROW(
+          auto aad,
+          key_value_metadata_->Get("row_group_aad_" + std::to_string(row_groups[0])));
+      out->set_file_decryptor_aad(aad);
+    }
```

Comment:
@pitrou regarding your suggestion to avoid storing AADs as `key_value_metadata` entries and instead read them from files at dataset scan time - do you have a suggestion on how/where to detect we'll no longer be reading `_metadata` file but rather data files? If we can do that we can change the decryptor's AAD there and read the desired data file.